package/MDAnalysis/auxiliary/base.py

```diff
@@ -0,0 +1,187 @@
+import numpy as np
+
+class AuxReader(object):
+    """ Base class for auxiliary readers
+    Does iteration and aligning between trajectory ts/auxiliary step
+    Actual reading/parsing of data to be handled by individual readers.
+    """
+      
+    # TODO deal with changing/different units
```

Comment:
Out of this PR scope, and even out of GSoC scope, but units got discussed in issue #596.